### PR TITLE
added documentation about live reloading

### PR DESCRIPTION
**Pull request by reflog**
#276

### CODE REVIEW

**Comment by vizo** [comment]
I tried but nodemon keeps restarting? I use linux..

**Comment by reflog** [comment]
Odd. I'm on Windows, it doesn't happen. Try to add '-V' to nodemon to see
which file causes the reload

**Comment by vizo** [comment]
debug value should be:
node ./build.js & nodemon --watch server --watch build.js --watch routes.js ./server.js
not:
nodemon ./build.js & nodemon --watch server --watch build.js --watch routes.js ./server.js

because if you call nodemon without --watch it is checking for all files in directory ...

**Comment by reflog** [comment]
absolutely correct. fixed.

**Comment by vizo** [comment]
But i think even this is not correct ... in this case it will not rebuild on changes ...

**Comment by reflog** [comment]
the only changes that won't be refreshed are the ones in build.js file. Which is pretty common in all projects... most server code goes into serve.js (and in my case /server folder which I added to watch)

**Comment by vizo** [comment]
I think build.js does build of other files, so it will never be rebuilt after changes of that files. But as i told you, i am not familiar with build process, i think @pragmadash can tell you more.

**Comment by vizo** [comment]
We should also consider about opt-out express livereload in release mode.

File: docs/index.md [hunk 1]
```diff
@@ -1140,6 +1140,31 @@ locator.registerInstance('browserifyPlugin', {
 * [catberry-debugger](https://github.com/catberry/catberry-debugger) – Chrome Dev Tools extension
 * [catberry-idea-plugin](https://github.com/catberry/catberry-idea-plugin) – Intellij platform plugin
 
+## Live Reloading
+
+Whilst Catberry doesn't have a built in live-reloading mechanism, it trivial to add it to your project by following these steps:
```
Comment (pragmadash) [inline docs/index.md]: I noticed a few typos:

* `built-in`
* `it's trivial`

File: docs/index.md [hunk 2]
```diff
@@ -1140,6 +1140,31 @@ locator.registerInstance('browserifyPlugin', {
...
+### Client side live reloading
+* `npm install -S express-livereload`
+* Modify `server.js` and place the following code below `const app = express();`:
+```
+var livereload = require('express-livereload');
+livereload(app);
+```  
+* Add the following line to `head.hbs`:
+```
+<script>document.write('<script src="http://' + (location.host || 'localhost').split(':')[0] + ':35729/livereload.js"></' + 'script>')</script>
```
Comment (pragmadash) [inline docs/index.md]: Do you have any idea why the highlighting is red here? Seems like a syntax error to me.

File: docs/index.md [hunk 3]
```diff
@@ -1140,6 +1140,31 @@ locator.registerInstance('browserifyPlugin', {
...
+	* `debug-win` command to: `powershell -Command \"Start-Process -NoNewWindow nodemon ./build.js; Start-Process -NoNewWindow -Wait  -ArgumentList '--watch server --watch build.js --watch routes.js  ./server.js' nodemon\"`
+	
+### Client side live reloading
+* `npm install -S express-livereload`
+* Modify `server.js` and place the following code below `const app = express();`:
+```
+var livereload = require('express-livereload');
+livereload(app);
+```  
+* Add the following line to `head.hbs`:
```
Comment (pragmadash) [inline docs/index.md]: I would say "to the head component's template". We have to be template-agnostic and don't mention any particular template engine.

File: docs/index.md (hunk 2 above)
Comment (pragmadash) [inline docs/index.md]: I'm wondering why we have to use this ugly `document.write` if we just could append a final `<script>` tag in the end of the document template.

Comment (reflog) [inline docs/index.md]: this is for the case where you are not using localhost.

File: docs/index.md (hunk 3 above)
Comment (reflog) [inline docs/index.md]: will fix

File: docs/index.md (hunk 1 above)
Comment (reflog) [inline docs/index.md]: will fix

**Comment by pragmadash** [comment]
@reflog I will include this to the next release, no worries. I would test the steps on my own before releasing these new docs.

**Pull request merged**